src/filehandler/pemreader.cpp

```diff
@@ -41,29 +41,20 @@ bool PemFileReader::checkFile() const
 
 Address PemFileReader::getAddress() const
 {
-    return Address(bytes(m_fileKeyBytes.begin() + crypto_sign_PUBLICKEYBYTES,m_fileKeyBytes.end()));
+    return Address(bytes(m_fileKeyBytes.begin() + PUBLIC_KEY_BYTES_LENGTH, m_fileKeyBytes.end()));
 }
 
 bytes PemFileReader::getSeed() const
 {
-    return bytes(m_fileKeyBytes.begin(),m_fileKeyBytes.begin() + crypto_sign_PUBLICKEYBYTES);
+    return bytes(m_fileKeyBytes.begin(), m_fileKeyBytes.begin() + PUBLIC_KEY_BYTES_LENGTH);
 }
 
 bytes PemFileReader::getPrivateKey() const
 {
-    unsigned char sk[crypto_sign_SECRETKEYBYTES];
-    unsigned char pk[crypto_sign_PUBLICKEYBYTES];
-    unsigned char seed[crypto_sign_SEEDBYTES];
+    bytes const seedBytes = getSeed();
+    bytes const pkBytes = getAddress().getPublicKey();
 
-    bytes seedBytes = getSeed();
-    bytes pkBytes = getAddress().getPublicKey();
-
-    std::copy(seedBytes.begin(), seedBytes.end(), seed);
-    std::copy(pkBytes.begin(), pkBytes.end(), pk);
-
-    crypto_sign_seed_keypair(pk, sk, seed);
-
-    return bytes(sk, sk + crypto_sign_SECRETKEYBYTES);
+    return wrapper::crypto::getSecretKey(pkBytes,seedBytes);
```

Comment:
Please have a look at:
https://github.com/ElrondNetwork/elrond-sdk-erdjs/blob/main/src/walletcore/userKeys.ts#L36
https://github.com/ElrondNetwork/elrond-sdk-erdpy/blob/main/erdpy/wallet/signing.py#L16

There, we only construct a keypair when signing.

Reply:
> Furthermore, why does crypto::getSecretKey() need the public key? (I will read below).

Because libsodium considers PRIVATE KEY = SEED + PUBLICKEY (64 bytes)
![image](https://user-images.githubusercontent.com/82832880/121533405-ebd3af80-ca08-11eb-828e-2e9ac95c751b.png)

Reply:
Update: We will use SECRET KEY = SEED + PUBLICKEY
When trying to initialize a Signer, it will expect a seed as constructor parameter and will compute the secret key out of it internally.